demos/fleet_provisioning/fleet_provisioning_csr/fleet_provisioning_csr_demo.c

```diff
@@ -0,0 +1,1248 @@
+ * placed in another file (mqtt_operations.c). This demo uses the coreMQTT
+ * library. If needed, mqtt_operations.c can be modified to replace coreMQTT
+ * with another MQTT library. This demo requires using the AWS IoT broker as
+ * Fleet Provisioning is an AWS IoT Core feature.
+ *
+ * This demo provisions a device certificate using the provisioning by claim
+ * workflow with a Certificate Signing Request (CSR). The demo connects to
+ * AWS IoT Core using provided claim credentials, subscribes to the
+ * CreateCertificateFromCsr topics, and obtains a certificate. It then subscribes
+ * to the RegisterThing topics and activates the certificate and obtains a Thing
+ * using the provisioning template. Finally, it reconnects to AWS IoT Core using
+ * the new credentials.
+ */
+
+/* Standard includes. */
+#include <stdlib.h>
+
+/* POSIX includes. */
+#include <unistd.h>
+
+/* Demo config. */
+#include "demo_config.h"
+
+/* MQTT operations. */
+#include "mqtt_operations.h"
+
+/* CBOR Library */
+#include "cbor.h"
+
+/* Fleet Provisioning Client Library. */
+#include "fleet_provisioning.h"
+
+/**
+ * These configurations are required. Throw compilation error if it is not
+ * defined.
+ */
+#ifndef PROVISIONING_TEMPLATE_NAME
+    #error "Please define PROVISIONING_TEMPLATE_NAME to the template name registered with AWS IoT Core in demo_config.h."
+#endif
+#ifndef CLAIM_CERT_PATH
+    #error "Please define path to claim certificate (CLAIM_CERT_PATH) in demo_config.h."
+#endif
+#ifndef CLAIM_KEY_PATH
+    #error "Please define path to claim private key (CLAIM_KEY_PATH) in demo_config.h."
+#endif
+#ifndef PROVISIONING_KEY_PATH
+    #error "Please define path to private key to provision (PROVISIONING_KEY_PATH) in demo_config.h."
+#endif
+#ifndef PROVISIONING_CSR_PATH
+    #error "Please define path to CSR to use to provision (PROVISIONING_CSR_PATH) in demo_config.h."
+#endif
+#ifndef PROVISIONING_CERT_PATH
+    #error "Please define path to which to write the obtained certificate (PROVISIONING_CERT_PATH) in demo_config.h."
+#endif
+
+/**
+ * @brief The length of #PROVISIONING_TEMPLATE_NAME.
+ */
+#define PROVISIONING_TEMPLATE_NAME_LENGTH    ( ( uint16_t ) ( sizeof( PROVISIONING_TEMPLATE_NAME ) - 1 ) )
+
+/**
+ * @brief Size of Thing name buffer.
+ */
+#define MAX_THING_NAME_LENGTH                30
+
+/**
+ * @brief Number of seconds to wait for the response from AWS IoT Fleet
+ * Provisioning service.
+ */
+#define FLEET_PROV_RESPONSE_WAIT_SECONDS     ( 2 )
+
+/**
+ * @brief The maximum number of times to run the loop in this demo.
+ *
+ * @note The demo loop is attempted to re-run only if it fails in an iteration.
+ * Once the demo loop succeeds in an iteration, the demo exits successfully.
+ */
+#ifndef FLEET_PROV_MAX_DEMO_LOOP_COUNT
+    #define FLEET_PROV_MAX_DEMO_LOOP_COUNT    ( 3 )
+#endif
+
+/**
+ * @brief Time in seconds to wait between retries of the demo loop if
+ * demo loop fails.
+ */
+#define DELAY_BETWEEN_DEMO_RETRY_ITERATIONS_S    ( 5 )
```

Comment:
This naming is inconsistent with `FLEET_PROV_RESPONSE_WAIT_SECONDS` macro for the `seconds` unit